src/operators.jl

```diff
@@ -87,9 +87,17 @@ if VERSION < v"0.6.0-dev.1632" # julia PR #17623
     @eval (.^)(lhs::Union{Variable,AffExpr}, rhs::Number) = lhs^rhs
 end
 # AffExpr--Variable
-(+){C,V<:JuMPTypes}(lhs::GenericAffExpr{C,V}, rhs::V) = (+)(rhs,lhs)
+(+){C,V<:JuMPTypes}(lhs::GenericAffExpr{C,V}, rhs::V) = GenericAffExpr{C,V}(vcat(lhs.vars,rhs),vcat(lhs.coeffs,one(C)), lhs.constant)
 (-){C,V<:JuMPTypes}(lhs::GenericAffExpr{C,V}, rhs::V) = GenericAffExpr{C,V}(vcat(lhs.vars,rhs),vcat(lhs.coeffs,-one(C)),lhs.constant)
-(*)(lhs::AffExpr, rhs::Variable) = (*)(rhs,lhs)
+# Don't fall back on AffExpr*Variable to preserve lhs/rhs consistency (appears in printing)
+function (*)(lhs::AffExpr, rhs::Variable)
+    n = length(lhs.vars)
+    if lhs.constant != 0.
```

Comment:
Could we do simply `lhs.constant * rhs` here and move the `if` to `(*)(lhs::Number, rhs::Variable)` ?

Reply:
How? What is `(*)(lhs::Number, rhs::Variable)` supposed to return if `lhs = 0`? An `AffExpr` with zero terms?

Reply:
Ok, `zero(AffExpr)` makes sense and is arguably simpler. However, I don't want to make arbitrary tweaks in performance critical code without profiling. This new method just copies Variable*AffExpr.